src/main/java/com/aws/greengrass/clientdevices/auth/iot/registry/ThingRegistry.java

```diff
@@ -62,16 +68,46 @@ public boolean isThingAttachedToCertificate(Thing thing, Certificate certificate
     }
 
     private void registerCertificateForThing(Thing thing, Certificate certificate) {
-        registry.put(thing.getThingName(), certificate.getIotCertificateId());
+        Set<CertificateEntry> certEntrySet = registry.computeIfAbsent(thing.getThingName(), s -> new HashSet<>());
+        // replace previous association with a new entry with extended TTL
+        clearThingCertificateAssociation(thing, certificate);
+        certEntrySet.add(getNewCertificateEntry(certificate));
+        registry.put(thing.getThingName(), certEntrySet);
```

Comment:
not needed? already taken care of from `computeIfAbsent` above

Reply:
right, not needed